app/database/operator/server_data_operator/transformers/group.ts

```diff
@@ -42,3 +44,29 @@ export const transformGroupRecord = ({action, database, value}: TransformerArgs)
         fieldsMapper,
     }) as Promise<GroupModel>;
 };
+
+/**
+   * transformGroupMembershipRecord: Prepares a record of the SERVER database 'GroupMembership' table for update or create actions.
+   * @param {TransformerArgs} operator
+   * @param {Database} operator.database
+   * @param {RecordPair} operator.value
+   * @returns {Promise<GroupMembershipModel>}
+   */
+export const transformGroupMembershipRecord = ({action, database, value}: TransformerArgs): Promise<GroupMembershipModel> => {
+    const raw = value.raw as GroupMembership;
+
+    // id of group comes from server response
+    const fieldsMapper = (model: GroupMembershipModel) => {
+        model._raw.id = raw.id || `${raw.group_id}-${raw.user_id}`;
```

Comment:
I saw this "id creation" at least twice in this PR (this file and in `server_data_operator/handlers/group.ts`). Should we create a function `getGroupId(groupId, userId) => '${groupId}-${userId}'` somewhere in utils and use that instead?

(notice I used single quotes instead of back ticks because it was messing with the formatting, so do not copy paste :P)